ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/Conversions.scala

```diff
@@ -246,28 +239,58 @@ private[state] object Conversions {
   /** Encodes a [[BlindingInfo]] into protobuf (i.e., [[DamlTransactionBlindingInfo]]).
     * It is consensus-safe because it does so deterministically.
     */
-  def encodeBlindingInfo(blindingInfo: BlindingInfo): DamlTransactionBlindingInfo =
+  def encodeBlindingInfo(
+      blindingInfo: BlindingInfo,
+      divulgedContracts: Map[ContractId, TransactionOuterClass.ContractInstance],
+  ): DamlTransactionBlindingInfo =
     DamlTransactionBlindingInfo.newBuilder
       .addAllDisclosures(encodeDisclosure(blindingInfo.disclosure).asJava)
-      .addAllDivulgences(encodeDivulgence(blindingInfo.divulgence).asJava)
+      .addAllDivulgences(encodeDivulgence(blindingInfo.divulgence, divulgedContracts).asJava)
       .build
 
-  def decodeBlindingInfo(blindingInfo: DamlTransactionBlindingInfo): BlindingInfo =
-    BlindingInfo(
-      disclosure = blindingInfo.getDisclosuresList.asScala.map { disclosureEntry =>
+  def decodeBlindingInfo(
+      damlTransactionBlindingInfo: DamlTransactionBlindingInfo
+  ): (BlindingInfo, Map[ContractId, Option[Value.ContractInst[VersionedValue[ContractId]]]]) = {
+    val divulgences = damlTransactionBlindingInfo.getDivulgencesList.asScala.map {
+      divulgenceEntry =>
+        val maybeContractInstance =
+          if (divulgenceEntry.hasContractInstance)
+            Some(decodeContractInstance(divulgenceEntry.getContractInstance))
+          else
+            None
+
+        val contractId = decodeContractId(
+          divulgenceEntry.getContractId
+        )
```

Comment:
Nit: this can be compacted on the same line, I think.

Reply:
Done